CMakeLists.txt

```diff
@@ -69,37 +68,23 @@ if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/lib/irrlichtmt")
 		message(FATAL_ERROR "IrrlichtMt project is missing a CMake target?!")
 	endif()
 
-	# set include dir the way it would normally be
-	set(IRRLICHT_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib/irrlichtmt/include")
-	set(IRRLICHT_LIBRARY IrrlichtMt)
 else()
-	find_package(Irrlicht)
-	if(BUILD_CLIENT AND NOT IRRLICHT_FOUND)
-		message(FATAL_ERROR "IrrlichtMt is required to build the client, but it was not found.")
-	elseif(NOT IRRLICHT_INCLUDE_DIR)
-		message(FATAL_ERROR "Irrlicht or IrrlichtMt headers are required to build the server, but none found.")
-	endif()
-endif()
+	find_package(IrrlichtMt QUIET)
+	if(NOT TARGET IrrlichtMt::IrrlichtMt)
+		if(BUILD_CLIENT)
+			message(FATAL_ERROR "IrrlichtMt is required to build the client, but it was not found.")
+		endif()
 
-include(CheckSymbolExists)
-set(CMAKE_REQUIRED_INCLUDES ${IRRLICHT_INCLUDE_DIR})
-unset(HAS_FORKED_IRRLICHT CACHE)
-check_symbol_exists(IRRLICHT_VERSION_MT "IrrCompileConfig.h" HAS_FORKED_IRRLICHT)
-if(NOT HAS_FORKED_IRRLICHT)
-	string(CONCAT EXPLANATION_MSG
-		"Irrlicht found, but it is not IrrlichtMt (Minetest's Irrlicht fork). "
-		"The Minetest team has forked Irrlicht to make their own customizations. "
-		"It can be found here: https://github.com/minetest/irrlicht")
-	if(BUILD_CLIENT)
-		message(FATAL_ERROR "${EXPLANATION_MSG}\n"
-			"Building the client with upstream Irrlicht is no longer possible.")
-	else()
-		message(WARNING "${EXPLANATION_MSG}\n"
-			"The server can still be built with upstream Irrlicht but this is DISCOURAGED.")
+		include(MinetestFindIrrlichtHeaders)
+		if(NOT IRRLICHT_INCLUDE_DIR)
+			message(FATAL_ERROR "Irrlicht or IrrlichtMt headers are required to build the server, but none found.")
+		endif()
+		message(STATUS "Building with old Irrlicht headers.")
```

Comment:
I don't think this message is needed.

Reply:
The reason I put it in is because otherwise there will be no message anywhere confirming that Irrlicht was found. Usually there'd be an "Irrlicht found" message with a version or something, so I thought providing that information under all circumstances might be useful. Do you want me to go ahead and remove the message anyway?

Reply:
Ah good point. I think `"Found Irrlicht headers: ${IRRLICHT_INCLUDE_DIR}"` would be good.